### PR TITLE
Travis for Python 3

### PR DESCRIPTION
**Pull request by ukoethe**

### CODE REVIEW

File: config/FindVIGRANUMPY_DEPENDENCIES.cmake [hunk 1]
```diff
@@ -2,65 +2,108 @@
...
+            CACHE FILEPATH "Python libraries"
+            FORCE)
+    ELSE()
+        IF(WIN32)
+            set(PYTHON_LIBRARY_NAME python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
+        ELSE()
+            execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
+                             "from distutils.sysconfig import *; print(get_config_var('LDLIBRARY'))"
+                              OUTPUT_VARIABLE PYTHON_LIBRARY_NAME OUTPUT_STRIP_TRAILING_WHITESPACE)
+            execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
```
Comment (jakirkham) [inline config/FindVIGRANUMPY_DEPENDENCIES.cmake]: Before this was run on Windows and Linux, which worked for me. However, now it appears this is just being run on Linux. I think this is what is causing AppVeyor to fail.

File: config/FindVIGRANUMPY_DEPENDENCIES.cmake [hunk 2]
```diff
@@ -2,65 +2,108 @@
...
+    #
+    #      find Python library
+    #
+    ######################################################################
+    IF(APPLE AND ${PYTHON_PREFIX} MATCHES ".*framework.*")
+        SET(PYTHON_LIBRARIES "${PYTHON_PREFIX}/Python"
+            CACHE FILEPATH "Python libraries"
+            FORCE)
+    ELSE()
+        IF(WIN32)
```
Comment (jakirkham) [inline config/FindVIGRANUMPY_DEPENDENCIES.cmake]: Regarding Windows, you know more than I do, but should something be done for Win64, as well? Would this work for both?

**Comment by jakirkham** [comment]
How have you configured AppVeyor? I was going to run a git bisect to see what went wrong, but I don't seem to have an `appveyor.yml` file to test with.

File: config/FindVIGRANUMPY_DEPENDENCIES.cmake (hunk 2 above)
Comment (ukoethe) [inline config/FindVIGRANUMPY_DEPENDENCIES.cmake]: According to my understanding of the cmake documentation (https://cmake.org/Wiki/CMake_Useful_Variables), WIN32 applies to all Windows versions.

Comment (jakirkham) [inline config/FindVIGRANUMPY_DEPENDENCIES.cmake]: Ah ok. Sorry, I was confused.

**Comment by ukoethe** [comment]
I think I configured AppVeyor directly on their website. It's much easier than the debug-by-pushing cycle of travis, but at the price that you can't see or edit it..

**Comment by ukoethe** [comment]
Shall we add Mac builds to the travis matrix as well?

**Comment by jakirkham** [comment]
Yeah, I like being able to have my CI configs in revision control. I made the less desirable choice of opening PRs against this to test instead.

Adding Mac builds sounds good.

File: config/FindVIGRANUMPY_DEPENDENCIES.cmake (hunk 1 above)
Comment (jakirkham) [inline config/FindVIGRANUMPY_DEPENDENCIES.cmake]: Turns out this is not the case and does nothing on Windows.

**Comment by jakirkham** [comment]
I am working on a Mac PR ( https://github.com/ukoethe/vigra/pull/303 ). It is not ready yet, but I will let you know when it is.

**Comment by jakirkham** [comment]
It might also be worth considering using the Ubuntu Trusty environment, as well. This would provide access to a newer version of Boost and hopefully allow testing against newer versions of Python.

**Comment by ukoethe** [comment]
Merged into `python3` branch.

**Comment by jakirkham** [comment]
Thanks.

**Pull request closed without merging**